components/nexus-core/src/main/java/org/sonatype/nexus/internal/orient/DatabaseServerImpl.java

```diff
@@ -76,10 +91,17 @@ public DatabaseServerImpl(final ApplicationDirectories applicationDirectories,
   }
 
   @ManagedAttribute
-  public boolean isListenerEnabled() {
-    return listenerEnabled;
+  public boolean isBinaryListenerEnabled() {
+    return binaryListenerEnabled;
+  }
+
+  @ManagedAttribute
+  public boolean isHttpListenerEnabled() {
+    return httpListenerEnabled;
   }
 
+  // FIXME: May need to revisit embedded configuration strategy, this is quite nasty
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)